Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm

```diff
@@ -1051,26 +1037,11 @@ @interface AVSampleBufferAudioRenderer (WebCoreSampleBufferKeySession) <AVConten
     updateAllRenderersHaveAvailableSamples();
 }
 
-void AudioVideoRendererAVFObjC::setHasRequestedAudioDataWhenReady(TrackIdentifier trackId, bool flag)
-{
-    DEBUG_LOG(LOGIDENTIFIER, "trackId", toString(trackId));
-    auto it = m_audioTracksMap.find(trackId);
-    if (it == m_audioTracksMap.end())
-        return;
-    auto& properties = it->value;
-    if (properties.hasRequestedAudioDataWhenReady == flag)
-        return;
-    DEBUG_LOG(LOGIDENTIFIER, flag);
-    properties.hasRequestedAudioDataWhenReady = flag;
-}
-
-bool AudioVideoRendererAVFObjC::hasRequestedAudioDataWhenReady(TrackIdentifier trackId)
+AudioVideoRendererAVFObjC::AudioTrackProperties& AudioVideoRendererAVFObjC::audioTrackPropertiesFor(TrackIdentifier trackId)
 {
-    DEBUG_LOG(LOGIDENTIFIER, "trackId", toString(trackId));
     auto it = m_audioTracksMap.find(trackId);
-    if (it == m_audioTracksMap.end())
-        return false;
-    return it->value.hasRequestedAudioDataWhenReady;
+    RELEASE_ASSERT(it != m_audioTracksMap.end());
```

Comment:
Are you certain that the given `trackId` should always be in the map, such that we'd crash here otherwise?
I see in the original code that that case was handled without asserting/crashing.